### PR TITLE
Genesis

### PR DESCRIPTION
**Pull request by nate-sys**
# Genesis
An almost mvp of weird.  

## Overview
Create and view pages which are generated incrementally using Perseus.
Absolutely no styling whatsoever, only simple HTML.

## Routes
Simply navigating to index should suffice. However, there are 3 routes
- `/` : The homepage
- `/create` : The form for creating a new page.
- `/link/<your username>` : The created page that is fetched from the database

## Notes
Currently all data is inserted and fetched from the database in a very crude and insecure way. This is just to init and demonstrate the app. This also doesn't implement editing the existing page.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,6 @@
-dist/
\ No newline at end of file
+dist/
+target/
+Cargo.lock
```
Comment (jbutterwick) [inline .gitignore:3]: Let's actually check in `Cargo.lock`
```suggestion

```

File: src/templates/create.rs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+use serde::{Deserialize, Serialize};
+use sycamore::prelude::*;
+
+// Only used for sycamore rendering
+#[derive(Serialize, Debug, Deserialize)]
+struct State {
+    #[serde(skip_serializing)]
+    pub username: RcSignal<String>,
+    pub name: RcSignal<String>,
+    pub links: RcSignal<Vec<InputLink>>,
```
Comment (jbutterwick) [inline src/templates/create.rs:11]: Let's get a field for "description" in here too

File: src/templates/links.rs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    headers.insert("DB", "weird".parse().unwrap());
+
+    // execute request
+    let client = reqwest::Client::new();
+    let response = client
+        .get(format!("{DB_URL}/key/user"))
+        .basic_auth("root", Some("root"))
+        .headers(headers)
+        .send()
+        .await?;
```
Comment (jbutterwick) [inline src/templates/links.rs:67]: When you use `?` you accept that your code will panic when approached with an error. In the future let's replace this with proper error handling

File: src/templates/links.rs [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    headers.insert("DB", "weird".parse().unwrap());
+
+    // execute request
+    let client = reqwest::Client::new();
+    let response = client
+        .get(format!("{DB_URL}/key/user/{username}"))
+        .basic_auth("root", Some("root"))
+        .headers(headers)
+        .send()
+        .await?;
```
Comment (jbutterwick) [inline src/templates/links.rs:104]: > When you use `?` you accept that your code will panic when approached with an error. In the future let's replace this with proper error handling

Same here

**Review by jbutterwick — APPROVED** [review]
I've made some comments, they're non-blocking and don't need to be resolved in this PR. 

You've done a great job with this, I'm very impressed 👍

LGTM

**Review by erlend-sh — APPROVED** [review]

**Pull request merged**